### PR TITLE
Dark mode 🌚

### PR DESCRIPTION
**Pull request by colebemis**
Implements dark mode using Theme UI.

### Dark 🌚 
image

### Light 🌞 

image

### CODE REVIEW

**Comment by johnletey** [comment]
@colebemis Is there a reason why you deleted the Carbon ads?

**Comment by colebemis** [comment]
I had a little Marie Kondo moment with the Carbon ad last night 😅 After thinking it through this morning, I think I'm going to add it back. Thanks for flagging that, @johnletey.

File: package.json [hunk 1]
```diff
@@ -15,30 +15,33 @@
...
+    "@mdx-js/react": "^1.5.1",
+    "@primer/primitives": "^2.0.0",
     "classnames": "2.2.5",
     "copy-to-clipboard": "^3.0.8",
     "downloadjs": "1.4.7",
     "feather-icons": "^4.24.1",
     "fuse.js": "3.2.0",
-    "gatsby": "^2.10.4",
+    "gatsby": "^2.17.10",
     "gatsby-plugin-emotion": "^4.1.0",
```
Comment (colebemis) [inline package.json]: I don't think I need the `emotion` plugin anymore.
```suggestion
```

File: src/components/Button.js [hunk 2]
```diff
@@ -1,21 +1,24 @@
...
   return (
     <Component
-      css={{
+      sx={{
         display: 'inline-block',
         margin: 0,
-        padding: `${theme.space[4]} ${theme.space[5]}`,
-        fontSize: theme.fontSizes[2],
-        fontWeight: theme.fontWeights.medium,
+        py: 4,
```
Comment (colebemis) [inline src/components/Button.js]:
```suggestion
        paddingY: 4,
```

File: src/components/Button.js [hunk 3]
```diff
@@ -1,21 +1,24 @@
...
+        px: 5,
```
Comment (colebemis) [inline src/components/Button.js]:
```suggestion
        paddingX: 5,
```

File: src/components/Footer.js [hunk 4]
```diff
@@ -1,23 +1,17 @@
...
-    <footer
-      css={{
-        display: 'flex',
-        flexDirection: 'column',
-        alignItems: 'center',
-        padding: theme.space[5],
-        fontSize: theme.fontSizes[1],
-      }}
-    >
+    <footer sx={{ py: 4, borderTop: '1px solid', borderColor: 'border' }}>
```
Comment (colebemis) [inline src/components/Footer.js]:
```suggestion
    <footer sx={{ paddingY: 4, borderTop: '1px solid', borderColor: 'border' }}>
```

File: src/components/Header.js [hunk 5]
```diff
@@ -1,99 +1,80 @@
...
+import { jsx } from 'theme-ui'
 import logOutboundLink from '../utils/logOutboundLink'
 import Link from './Link'
 import OutboundLink from './OutboundLink'
+import { useColorMode } from 'theme-ui'
+import Icon from './Icon'
+import Button from './Button'
 
 function Header() {
+  const [colorMode, setColorMode] = useColorMode()
```
Comment (colebemis) [inline src/components/Header.js]: Let's move this into a separate `ColorModeToggle`component.

File: src/components/Header.js [hunk 6]
```diff
@@ -1,99 +1,80 @@
...
   return (
-    <div
-      css={{
-        padding: `${theme.space[4]} ${theme.space[5]}`,
+    <header
+      sx={{
+        py: 4,
```
Comment (colebemis) [inline src/components/Header.js]:
```suggestion
        paddingY: 4,
```

File: src/components/Header.js [hunk 7]
```diff
@@ -1,99 +1,80 @@
...
+        px: 5,
```
Comment (colebemis) [inline src/components/Header.js]:
```suggestion
        paddingX: 5,
```

File: src/components/Header.js [hunk 8]
```diff
@@ -1,99 +1,80 @@
...
-            backgroundColor: theme.colors.gray[2],
+            bg: 'background',
           },
         }}
       >
         v{version}
       </OutboundLink>
 
-      <div css={{ margin: '0 auto' }} />
+      <div sx={{ mx: 'auto' }} />
```
Comment (colebemis) [inline src/components/Header.js]:
```suggestion
      {/* Spacer */}
      <div sx={{ mx: 'auto' }} />
```

File: src/components/Hero.js [hunk 9]
```diff
@@ -1,89 +1,73 @@
...
       }}
     >
       <h1
-        css={{
-          margin: `0 0 ${theme.space[5]}`,
-          fontSize: theme.fontSizes[6],
-          fontWeight: theme.fontWeights.normal,
-          lineHeight: theme.lineHeights.tight,
+        sx={{
+          mt: 0,
```
Comment (colebemis) [inline src/components/Hero.js]:
```suggestion
          marginTop: 0,
```

File: src/components/Hero.js [hunk 10]
```diff
@@ -1,89 +1,73 @@
...
+          mb: 5,
```
Comment (colebemis) [inline src/components/Hero.js]:
```suggestion
          marginBottom: 5,
```

File: src/components/IconTile.js [hunk 11]
```diff
@@ -49,10 +49,10 @@ function IconTile({ name, onClick, ...props }) {
         </div>
       </AspectRatioBox>
       <span
-        css={{
-          fontSize: theme.fontSizes[1],
-          color: theme.colors.gray[8],
-          padding: ` 0 ${theme.space[4]} ${theme.space[4]}`,
+        sx={{
+          fontSize: 1,
+          px: 4,
```
Comment (colebemis) [inline src/components/IconTile.js]:
```suggestion
          paddingX: 4,
```

File: src/components/IconTile.js [hunk 12]
```diff
@@ -49,10 +49,10 @@ function IconTile({ name, onClick, ...props }) {
...
+          pb: 4,
```
Comment (colebemis) [inline src/components/IconTile.js]:
```suggestion
          paddingBottom: 4,
```

File: src/components/SearchInput.js [hunk 13]
```diff
@@ -22,45 +22,45 @@ function SearchInput({ placeholder, value, onChange, ...props }) {
...
+          padding: 4,
+          paddingLeft: 52,
           fontSize: 'inherit',
-          lineHeight: theme.lineHeights.none,
+          lineHeight: 'none',
           fontFamily: 'inherit',
           color: 'inherit',
-          background: theme.colors.white,
-          boxShadow: theme.shadows[1],
+          bg: 'white',
```
Comment (colebemis) [inline src/components/SearchInput.js]:
```suggestion
          backgroundColor: 'background',
```

File: src/components/SearchInput.js [hunk 14]
```diff
@@ -22,45 +22,45 @@ function SearchInput({ placeholder, value, onChange, ...props }) {
...
+          boxShadow: 1,
           border: 0,
-          borderRadius: 0,
           appearance: 'none',
           outline: 0,
-          borderRadius: theme.radii[1],
+          bg: 'background',
```
Comment (colebemis) [inline src/components/SearchInput.js]:
```suggestion
```

File: src/components/CarbonAd.js [hunk 15]
```diff
@@ -1,52 +0,0 @@
-import React from 'react'
-import theme from '../theme'
-
-function CarbonAd(props) {
```
Comment (colebemis) [inline src/components/CarbonAd.js]: Let's add this back 😅

File: src/components/Header.js [hunk 16]
```diff
@@ -1,99 +1,80 @@
...
 import { Link as GatsbyLink } from 'gatsby'
-import MediaQuery from 'react-responsive'
-import theme from '../theme'
+import { jsx } from 'theme-ui'
 import logOutboundLink from '../utils/logOutboundLink'
 import Link from './Link'
 import OutboundLink from './OutboundLink'
+import { useColorMode } from 'theme-ui'
+import Icon from './Icon'
+import Button from './Button'
```
Comment (colebemis) [inline src/components/Header.js]: Imports need to be reordered here.

File: src/components/Header.js [hunk 17]
```diff
@@ -1,99 +1,80 @@
...
         href="https://github.com/feathericons/feather/releases"
         onClick={() => logOutboundLink('release notes')}
-        css={{
-          padding: `${theme.space[1]} ${theme.space[2]}`,
-          marginLeft: theme.space[2],
-          fontSize: theme.fontSizes[1],
-          lineHeight: theme.lineHeights.tight,
-          fontWeight: theme.fontWeights.medium,
+        sx={{
+          py: 1,
```
Comment (colebemis) [inline src/components/Header.js]:
```suggestion
          paddingY: 1,
```

File: src/components/Header.js [hunk 18]
```diff
@@ -1,99 +1,80 @@
...
+          px: 2,
```
Comment (colebemis) [inline src/components/Header.js]:
```suggestion
          paddingX: 2,
```

File: src/components/Header.js [hunk 19]
```diff
@@ -1,99 +1,80 @@
...
+          ml: 2,
```
Comment (colebemis) [inline src/components/Header.js]:
```suggestion
          marginLeft: 2,
```

File: src/components/Header.js [hunk 20]
```diff
@@ -1,99 +1,80 @@
...
-          color: theme.colors.gray[7],
-          backgroundColor: theme.colors.gray[1],
-          borderRadius: theme.radii[1],
+          color: 'inherit',
+          border: '1px solid',
+          borderColor: 'border',
+          borderRadius: 1,
           '&:hover': {
-            backgroundColor: theme.colors.gray[2],
+            bg: 'background',
```
Comment (colebemis) [inline src/components/Header.js]:
```suggestion
            backgroundColor: 'background',
```

File: src/components/Header.js [hunk 21]
```diff
@@ -1,99 +1,80 @@
...
-          GitHub
-        </Link>
-      </div>
-    </div>
+      <Button
+        onClick={e => {
+          setColorMode(colorMode === 'default' ? 'dark' : 'default')
+        }}
+        sx={{
+          bg: 'transparent',
```
Comment (colebemis) [inline src/components/Header.js]:
```suggestion
          backgroundColor: 'transparent',
```

File: src/components/Header.js [hunk 22]
```diff
@@ -1,99 +1,80 @@
...
+          p: 1,
```
Comment (colebemis) [inline src/components/Header.js]:
```suggestion
          padding: 1,
```

File: src/components/Header.js [hunk 23]
```diff
@@ -1,99 +1,80 @@
...
+          color: 'icon',
+          ml: 4,
```
Comment (colebemis) [inline src/components/Header.js]:
```suggestion
          marginLeft: 4,
```

File: src/components/Hero.js [hunk 24]
```diff
@@ -1,89 +1,73 @@
...
         display: 'flex',
         flexDirection: 'column',
         alignItems: 'center',
-        padding: `${theme.space[8]} ${theme.space[5]} ${theme.space[9]}`,
-        backgroundColor: theme.colors.white,
-        borderBottom: `1px solid ${theme.colors.gray[2]}`,
-        [mediaQuery(theme.breakpoints.small)]: {
-          padding: `${theme.space[9]} ${theme.space[5]} ${theme.space[10]}`,
-        },
+        py: [8, 9],
```
Comment (colebemis) [inline src/components/Hero.js]:
```suggestion
        paddingY: [8, 9],
```

File: src/components/Hero.js [hunk 25]
```diff
@@ -1,89 +1,73 @@
...
+        px: 5,
```
Comment (colebemis) [inline src/components/Hero.js]:
```suggestion
        paddingX: 5,
```

File: src/components/Header.js [hunk 26]
```diff
@@ -1,99 +1,80 @@
...
-          as={OutboundLink}
-          href="https://github.com/feathericons/feather"
-          onClick={() => logOutboundLink('github')}
-        >
-          GitHub
-        </Link>
-      </div>
-    </div>
+      <Button
+        onClick={e => {
```
Comment (colebemis) [inline src/components/Header.js]:
```suggestion
        onClick={() => {
```

File: src/components/IconTile.js [hunk 27]
```diff
@@ -16,28 +16,28 @@ function IconTile({ name, onClick, ...props }) {
...
         }
       }}
-      css={{
+      sx={{
         display: 'flex',
         flexDirection: 'column',
         textAlign: 'center',
         alignItems: 'center',
         cursor: 'pointer',
+        bg: 'background',
```
Comment (colebemis) [inline src/components/IconTile.js]:
```suggestion
        backgroundColor: 'background',
```

File: src/gatsby-plugin-theme-ui/index.js [hunk 28]
```diff
@@ -0,0 +1,51 @@
+import { colors } from '@primer/primitives'
+
+export default {
```
Comment (colebemis) [inline src/gatsby-plugin-theme-ui/index.js]: I need to remove the old `theme.js` file.

File: src/pages/index.js [hunk 29]
```diff
@@ -23,30 +22,40 @@ function IndexPage({ location }) {
...
-        css={{
-          position: 'sticky',
-          top: 0,
-          boxShadow: `inset 0 16px 16px ${theme.colors.gray[0]}`,
-          zIndex: 1,
+      <div
+        sx={{
+          display: 'grid',
+          gridGap: 6,
+          py: 4,
```
Comment (colebemis) [inline src/pages/index.js]:
```suggestion
          paddingY: 4,
```

File: src/pages/index.js [hunk 30]
```diff
@@ -23,30 +22,40 @@ function IndexPage({ location }) {
...
+          px: 5,
```
Comment (colebemis) [inline src/pages/index.js]:
```suggestion
          paddingX: 5,
```

File: src/pages/index.js [hunk 31]
```diff
@@ -23,30 +22,40 @@ function IndexPage({ location }) {
...
+          mx: 'auto',
```
Comment (colebemis) [inline src/pages/index.js]:
```suggestion
          marginX: 'auto',
```

File: src/pages/index.js [hunk 32]
```diff
@@ -23,30 +22,40 @@ function IndexPage({ location }) {
...
-          value={query || ''}
-          onChange={event => setQuery(event.target.value)}
-        />
-      </Container>
-      <Container>
+        <div
+          sx={{
+            position: 'sticky',
+            top: 0,
+            pt: 4,
```
Comment (colebemis) [inline src/pages/index.js]:
```suggestion
            paddingTop: 4,
```

File: src/pages/404.js [hunk 33]
```diff
@@ -1,34 +1,26 @@
...
-          display: 'flex',
-          flexDirection: 'column',
-          alignItems: 'center',
-          padding: `${theme.space[9]} ${theme.space[5]}`,
-          borderTop: `1px solid ${theme.colors.gray[2]}`,
+      <h1
+        sx={{
+          py: 9,
+          px: 5,
+          m: 0,
```
Comment (colebemis) [inline src/pages/404.js]:
```suggestion
          margin: 0,
```

File: src/pages/404.js [hunk 34]
```diff
@@ -1,34 +1,26 @@
...
-        css={{
-          display: 'flex',
-          flexDirection: 'column',
-          alignItems: 'center',
-          padding: `${theme.space[9]} ${theme.space[5]}`,
-          borderTop: `1px solid ${theme.colors.gray[2]}`,
+      <h1
+        sx={{
+          py: 9,
+          px: 5,
```
Comment (colebemis) [inline src/pages/404.js]:
```suggestion
          paddingX: 5,
```

File: src/pages/404.js [hunk 35]
```diff
@@ -1,34 +1,26 @@
...
-      <div
-        css={{
-          display: 'flex',
-          flexDirection: 'column',
-          alignItems: 'center',
-          padding: `${theme.space[9]} ${theme.space[5]}`,
-          borderTop: `1px solid ${theme.colors.gray[2]}`,
+      <h1
+        sx={{
+          py: 9,
```
Comment (colebemis) [inline src/pages/404.js]:
```suggestion
          paddingY: 9,
```

File: src/components/Hero.js [hunk 36]
```diff
@@ -1,89 +1,73 @@
...
           onClick={() => logOutboundLink('get started')}
-          css={{
-            color: theme.colors.white,
-            backgroundColor: theme.colors.primary[0],
-            '&:hover': {
-              backgroundColor: theme.colors.primary[1],
-            },
+          sx={{
+            color: 'white',
+            bg: 'primary',
```
Comment (colebemis) [inline src/components/Hero.js]:
```suggestion
            backgroundColor: 'primary',
```

File: src/components/Hero.js [hunk 37]
```diff
@@ -1,89 +1,73 @@
...
-          css={{
-            color: theme.colors.gray[8],
-            backgroundColor: 'transparent',
-            boxShadow: `inset 0 0 0 1px ${theme.colors.gray[4]}`,
+          sx={{
+            bg: 'transparent',
+            boxShadow: theme => `inset 0 0 0 1px ${theme.colors.border}`,
             '&:hover': {
-              backgroundColor: theme.colors.gray[0],
+              bg: 'background',
```
Comment (colebemis) [inline src/components/Hero.js]:
```suggestion
              backgroundColor: 'background',
```

**Pull request merged**